### PR TITLE
Beta 0.20

### PR DESCRIPTION
**Pull request by jkrumbiegel**
This branch contains changes for the upcoming version v0.20, which we want people to try out before fully switching (therefore the "beta" status).

This branch contains the high-dpi work for GLMakie and the ability to show correctly scaled png images in web contexts such as Pluto, VSCode, Jupyter, Quarto etc. Because these changes eliminate the disparity between Mac and the other OSes, we can finally make adjustments to the general theme to account for that. So far, window size was a compromise so it wouldn't be too small on Mac and too large on Windows/Linux (at standard dpi values). Now we scale it down so it is more useable across a wide range of contexts, whether in documents or in interactive windows.

TODOs:
- [x] Antialiasing of lines scales with `px_per_unit` in GLMakie, making lines more blurry than necessary

### CODE REVIEW

**Comment by ffreyer** [comment]
#2746 and #2598 should also be in v0.20 imo

**Comment by jkrumbiegel** [comment]
@ffreyer the mesh one is not marked breaking, couldn't that be merged in anytime?

**Comment by SimonDanisch** [comment]
I would love to make it none breaking, but right now it's getting discussed, if we want to change the sampling position of the underlying data...

File: GLMakie/test/unit_tests.jl [hunk 1]
```diff
@@ -255,3 +257,131 @@ end
...
+
+    # render at lower resolution
+    screen = display(GLMakie.Screen(visible = false, scalefactor = 2, px_per_unit = 1), fig)
+    @test screen.scalefactor[] === 2f0
+    @test screen.px_per_unit[] === 1f0
+    @test size(screen.framebuffer) == (W, H)
+
+    # decrease the scale factor after-the-fact
+    screen.scalefactor[] = 1
+    @test GLMakie.window_size(screen.glscreen) == scaled(screen, (W, H))
```
Comment (ffreyer) [inline GLMakie/test/unit_tests.jl:309]: This fails for me on ubuntu
```
  Expression: GLMakie.window_size(screen.glscreen) == scaled(screen, (W, H))
   Evaluated: (800, 800) == (400, 400)
```

Comment (ffreyer) [reply GLMakie/test/unit_tests.jl:309]: Running it manually works though. Just not in `]test GLMakie`...?

Comment (ffreyer) [reply GLMakie/test/unit_tests.jl:310]: No it doesn't. It's failing again after merging another pr locally

**Comment by ffreyer** [comment]
Closes #2598

I tried fixing the line AA stuff but I'm not sure if I have the correct idea on how it should look. I basically tried to match lines with half resolution and linewidth at 200% zoom with `px_per_unit = 0.5`.

<details>

<summary>Examples</summary>

Screenshot from 2023-07-12 16-58-15
Screenshot from 2023-07-12 17-23-57
Screenshot from 2023-07-12 17-24-23
Screenshot from 2023-07-12 17-24-33
Screenshot from 2023-07-12 17-25-35
Screenshot from 2023-07-12 17-25-44

</details>

**Comment by jkrumbiegel** [comment]
@ffreyer line antialiasing looks perfect now, thanks! Here with `px_per_unit = 10`:

grafik

**Comment by ffreyer** [comment]
Also closes #2522?

**Comment by SimonDanisch** [comment]
Closing in favor of: https://github.com/MakieOrg/Makie.jl/pull/3113

**Pull request closed without merging**